course/include/isometry.h

```diff
@@ -0,0 +1,83 @@
+#ifndef ISOMETRY_H
+#define	ISOMETRY_H
+
+// Standard libraries
+#include <cmath>
+#include <initializer_list>
+#include <iostream>
+#include <limits>
+#include <sstream>
+#include <string>
+
+namespace ekumen {
+namespace math {
+
+static constexpr int resolution{4};
+template <class T>
+typename std::enable_if<!std::numeric_limits<T>::is_integer, bool>::type almost_equal(T a, T b, int resolution) {
+    return std::fabs(a - b) <= std::numeric_limits<T>::epsilon() * std::fabs(a + b) * resolution
+           || std::fabs(a - b) < std::numeric_limits<T>::min();
+}
+
+class Vector3 {
+    public:
+        Vector3(const double &x = 0, const double &y = 0, const double &z = 0);
+        ~Vector3();
+
+        // Getter for elements of the vector.
+        const double& x() const { return this->x_; };
+        const double& y() const { return this->y_; };
+        const double& z() const { return this->z_; };
+
+        // Setter for elements of the vector.
+        void x(const double &x) { this->x_ = x; };
+        void y(const double &y) { this->y_ = y; };
+        void z(const double &z) { this->z_ = z; };
```

Comment:
These setters seem redundant given that we already have the `x()` and friends accessors.